pkg/streamingpromql/benchmarks/comparison_test.go

```diff
@@ -111,7 +112,7 @@ func TestBothEnginesReturnSameResultsForBenchmarkQueries(t *testing.T) {
 			start := time.Unix(int64((NumIntervals-c.Steps)*intervalSeconds), 0)
 			end := time.Unix(int64(NumIntervals*intervalSeconds), 0)
 
-			prometheusResult, prometheusClose := c.Run(ctx, t, start, end, interval, prometheusEngine, q)
+			prometheusResult, prometheusClose := c.Run(limiter.ContextWithNewUnlimitedMemoryConsumptionTracker(ctx), t, start, end, interval, prometheusEngine, q)
```

Comment:
Could we use a `UnlimitedMemoryTrackerPromqlEngine` instance here (and in all other tests that create an instance of Prometheus' engine in `pkg/streamingpromql`)? Then we don't need to do this.

Reply:
I meant to use `UnlimitedMemoryTrackerPromqlEngine` for engine but instead used `ContextWithNewUnlimitedMemoryConsumptionTracker` in context. 

Fixed in 711fbdde234d26fed1de9fe94589ebd1105bdadc

For other test, if they don't touch both ingester/store-gateway response layer and streamingpromql operator (which only applies in the labels memory tracking case), I think we can just leave the prometheusEngine used in non-breaking test as it is.